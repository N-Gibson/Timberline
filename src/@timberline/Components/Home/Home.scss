.section_container {
  display: flex;
  align-items: center;
  justify-content: space-around;
  padding: 5em 0;
}

.banner {
  background-image: url('../../assets/Venado_front.jpg');
  background-repeat: no-repeat;
  background-attachment: fixed;
  height: 85vh;
  background-position: center;
  background-size: cover;
  display: flex;
  justify-content: center;
  flex-direction: column;
  margin-top: 120px;
  width: 100vw;
}

.banner * {
  margin: 0.5em 0;
}

.logo_text_background {
  background-image: url('../../assets/translucent.png');
  background-repeat: no-repeat;
  background-position: center;
  border-radius: 10px;
  display: flex;
  justify-content: center;
  margin: auto;
  padding: 15px 0px;
  position: relative;
  width: 75%;
}

.article_container {
  width: 33%;
  border-radius: 6px;
  background-color: #c8ad96;
  box-shadow: 10px 0px 20px 0px #00000040;
  padding: 1.5em;
}

.section_img {
  border-radius: 16px;
  background-color: #c8ad96;
  box-shadow: 10px 0px 20px 0px #00000040;
}

.article_header {
  margin: 0;
  word-spacing: 2px;
}

.article_text {
  letter-spacing: 1px;
  line-height: 2;
  font-size: 1.25em;
}

#home_center_main {
  background-color: #e0e0e0;
  box-shadow: 10px 0px 20px 0px #00000040;
  margin: 0 0 5em 0;
}

.section_contact_form {
  align-items: center;
  display: flex;
  flex-direction: column;
}

li {
  list-style-type: none;
}

#rrr_img {
  width: 40%;
}

#meet_owner_img {
  width: 30%;
}

@media screen and (max-width: 1850px) {
  #timberline_text_only {
    width: 1000px;
  }
}

@media screen and (max-width: 1400px) {
  #timberline_text_only {
    width: 800px;
  }
}

@media screen and (max-width: 1100px) {
  #timberline_text_only {
    width: 600px;
  }
}
