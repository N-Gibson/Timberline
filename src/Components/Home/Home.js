import React, { Component } from 'react';
import ImageGallery from 'react-image-gallery';
import venadoFront from '../../assets/Venado_front.jpg';
import purpleAster from '../../assets/Purple_Aster.jpg';
import profile from '../../assets/profile.jpg';
import "react-image-gallery/styles/css/image-gallery.css";

export class Home extends Component {
  constructor() {
    super()
    this.state = {
      showThumbnails: false,
      showFullscreenButton: false,
      showPlayButton: false,
      showBullets: true,
      autoPlay: true,
      slideInterval: 5000,
      images: [
        {
          original: venadoFront,
          thumbnail: venadoFront,
          originalClass: 'venado_gallery_image',
          originalAlt: 'Custom home in Santa Fe, New Mexico. Rear view of home.',
        },
        {
          original: purpleAster,
          thumbnail: purpleAster,
          originalClass: 'purpleAster_gallery_image',
          originalAlt: 'Custom home in Santa Fe, New Mexico. Mountain views.',
        },
      ],
    }
  }

  render() {
    return (
      <section className="home">
        <ImageGallery 
          items={this.state.images} 
          showThumbnails={this.state.showThumbnails}
          showFullscreenButton={this.state.showFullscreenButton}
          showPlayButton={this.state.showPlayButton}
          showBullets={this.state.showBullets}
          autoPlay={this.state.autoPlay}
          slideInterval={this.state.slideInterval}
        />
        <div className="center_main">
          <div className="rrr_content">
            <h2>Raise, Repair, Remodel</h2>
            <p>Timberline Inc. is a locally owned and operated business in Santa Fe, New Mexico and surrounding areas since 1992. Specializing in New Mexican style builds and remodels, Timberline can do it all. From the preservation of original homes dating back to the mid 1800's to brand new homes and everything in between! Award winning Timberline Inc. can help with repairs, remodels or raising a new project from the ground up.</p>
          </div>
          <img src={purpleAster}/>
        </div>
        <div className="bottom_main">
          <h2>Meet the Owner</h2>
          <p>Stephen Gibson, owner and founder of Timberline Inc. has been working in the Santa Fe community for over 35 years and has received 11 awards in the Santa Fe area Home Builders Association's - A Parade of Homes. Steve's passion is crafting quality solutions for your construction needs.</p>
          <img src={profile}/>
        </div>
      </section>
    )
  }
}
