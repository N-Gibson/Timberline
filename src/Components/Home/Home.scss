.section_container {
  display: flex;
  align-items: center;
  justify-content: space-around;
  margin: 5em 0;
  padding: 5em 0;
}

.article_container {
  width: 33%;
  border-radius: 6px;
  background-color: #C8AD96;
  box-shadow: 10px 0px 20px 0px #00000040;
  padding: 1.5em;
}

.section_img {
  border-radius: 16px;
  background-color: #C8AD96;
  box-shadow: 10px 0px 20px 0px #00000040;
}

.article_header {
  margin: 0;
  word-spacing: 2px;
}

.article_text {
  letter-spacing: 1px;
  line-height: 2;
  font-size: 1.25em;
}

#home_center_main {
  background-color: #E0E0E0;
  box-shadow: 10px 0px 20px 0px #00000040;
  margin: 0 0 5em 0;
}

#rrr_img {
  width: 40%;
}

#meet_owner_img {
  width: 30%;
}
