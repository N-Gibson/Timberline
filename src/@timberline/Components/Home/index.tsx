import React from 'react'
import ScrollAnimation from 'react-animate-on-scroll'

import purpleAster from '@timberline/assets/Purple_Aster.jpg'
import profile from '@timberline/assets/profile.jpg'
import timberline_text_only from '@timberline/assets/timberline_text_only.png'

import './Home.scss'

interface HomeProps {}

const Home: React.FC<HomeProps> = () => {
  return (
    <section className="home">
      <div className="banner_container">
        <div className="banner">
          <ScrollAnimation animateIn="fadeIn">
            <div className="logo_text_background">
              <img src={timberline_text_only} alt="timberline logo text"></img>
            </div>
          </ScrollAnimation>
        </div>
      </div>
      <section id="home_center_main" className="section_container">
        <article className="article_container">
          <h2 className="article_header">Raise, Repair, Remodel</h2>
          <p className="article_text">
            Timberline Inc. is a locally owned and operated business in Santa
            Fe, New Mexico and surrounding areas since 1992. Specializing in New
            Mexican style builds and remodels, Timberline can do it all. From
            the preservation of original homes dating back to the mid 1800's to
            brand new homes and everything in between! Award winning Timberline
            Inc. can help with repairs, remodels or raising a new project from
            the ground up.
          </p>
        </article>
        <img
          id="rrr_img"
          className="section_img"
          src={purpleAster}
          alt="house in valley"
        />
      </section>
      <section className="section_container" id="home_bottom_main">
        <img
          id="meet_owner_img"
          className="section_img"
          src={profile}
          alt="timberline owner"
        />
        <article className="article_container">
          <h2 className="article_header">Meet the Owner</h2>
          <p className="article_text">
            Stephen Gibson, owner and founder of Timberline Inc. has been
            working in the Santa Fe community for over 35 years and has received
            11 awards in the Santa Fe area Home Builders Association's - A
            Parade of Homes. Steve's passion is crafting quality solutions for
            your construction needs.
          </p>
        </article>
      </section>
    </section>
  )
}

export { Home }
