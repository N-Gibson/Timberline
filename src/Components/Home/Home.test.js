import React from 'react';
import { shallow } from 'enzyme';
import './Home';

describe('Home', () => {
  it('should have a passing test for now', () => {
    expect(true).toEqual(true);
  });
});