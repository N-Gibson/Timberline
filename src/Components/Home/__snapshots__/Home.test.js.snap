// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Home component should match snapshot 1`] = `
<section
  className="home"
>
  <ImageGallery
    additionalClass=""
    autoPlay={true}
    disableKeyDown={false}
    disableSwipe={false}
    disableThumbnailScroll={false}
    flickThreshold={0.4}
    indexSeparator=" / "
    infinite={true}
    isRTL={false}
    items={
      Array [
        Object {
          "original": "Venado_front.jpg",
          "originalAlt": "Custom home in Santa Fe, New Mexico. Rear view of home.",
          "originalClass": "venado_gallery_image",
          "thumbnail": "Venado_front.jpg",
        },
        Object {
          "original": "Purple_Aster.jpg",
          "originalAlt": "Custom home in Santa Fe, New Mexico. Mountain views.",
          "originalClass": "purpleAster_gallery_image",
          "thumbnail": "Purple_Aster.jpg",
        },
      ]
    }
    lazyLoad={false}
    onBeforeSlide={null}
    onClick={null}
    onErrorImageURL=""
    onImageError={null}
    onImageLoad={null}
    onMouseLeave={null}
    onMouseOver={null}
    onPause={null}
    onPlay={null}
    onScreenChange={null}
    onSlide={null}
    onThumbnailClick={null}
    onThumbnailError={null}
    onTouchEnd={null}
    onTouchMove={null}
    onTouchStart={null}
    preventDefaultTouchmoveEvent={false}
    renderCustomControls={null}
    renderFullscreenButton={[Function]}
    renderItem={null}
    renderLeftNav={[Function]}
    renderPlayPauseButton={[Function]}
    renderRightNav={[Function]}
    renderThumbInner={null}
    showBullets={true}
    showFullscreenButton={false}
    showIndex={false}
    showNav={true}
    showPlayButton={false}
    showThumbnails={false}
    slideDuration={450}
    slideInterval={5000}
    slideOnThumbnailOver={false}
    startIndex={0}
    stopPropagation={false}
    swipeThreshold={30}
    swipingTransitionDuration={0}
    thumbnailPosition="bottom"
    useBrowserFullscreen={true}
    useTranslate3D={true}
  />
  <div
    className="center_main"
  >
    <div
      className="rrr_content"
    >
      <h2>
        Raise, Repair, Remodel
      </h2>
      <p>
        Timberline Inc. is a locally owned and operated business in Santa Fe, New Mexico and surrounding areas since 1992. Specializing in New Mexican style builds and remodels, Timberline can do it all. From the preservation of original homes dating back to the mid 1800's to brand new homes and everything in between! Award winning Timberline Inc. can help with repairs, remodels or raising a new project from the ground up.
      </p>
    </div>
    <img
      alt="house in valley"
      src="Purple_Aster.jpg"
    />
  </div>
  <div
    className="bottom_main"
  >
    <h2>
      Meet the Owner
    </h2>
    <p>
      Stephen Gibson, owner and founder of Timberline Inc. has been working in the Santa Fe community for over 35 years and has received 11 awards in the Santa Fe area Home Builders Association's - A Parade of Homes. Steve's passion is crafting quality solutions for your construction needs.
    </p>
    <img
      alt="timberline owner"
      src="profile.jpg"
    />
  </div>
</section>
`;
